fastly/block_fastly_service_v1_logging_honeycomb_test.go

```diff
@@ -0,0 +1,276 @@
+	domain := fmt.Sprintf("fastly-test.%s.com", name)
+
+	log1 := gofastly.Honeycomb{
+		Version:       1,
+		Name:          "honeycomb-endpoint",
+		Token:         "s3cr3t",
+		Dataset:       "dataset",
+		FormatVersion: 2,
+		Format:        appendNewLine(honeycombDefaultFormat),
+	}
+
+	log1_after_update := gofastly.Honeycomb{
+		Version:           1,
+		Name:              "honeycomb-endpoint",
+		Dataset:           "new-dataset",
+		Token:             "secret",
+		FormatVersion:     2,
+		Format:            appendNewLine(honeycombDefaultFormat),
+		Placement:         "none",
+		ResponseCondition: "response_condition_test",
+	}
+
+	log2 := gofastly.Honeycomb{
+		Version:           1,
+		Name:              "another-honeycomb-endpoint",
+		Token:             "another-token",
+		Dataset:           "another-dataset",
+		FormatVersion:     2,
+		Format:            appendNewLine(honeycombDefaultFormat),
+		Placement:         "none",
+		ResponseCondition: "response_condition_test",
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckServiceV1Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceV1HoneycombConfig(name, domain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceV1HoneycombAttributes(&service, []*gofastly.Honeycomb{&log1}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "name", name),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "logging_honeycomb.#", "1"),
+				),
+			},
+
+			{
+				Config: testAccServiceV1HoneycombConfig_update(name, domain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceV1HoneycombAttributes(&service, []*gofastly.Honeycomb{&log1_after_update, &log2}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "name", name),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "logging_honeycomb.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckFastlyServiceV1HoneycombAttributes(service *gofastly.ServiceDetail, honeycomb []*gofastly.Honeycomb) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		conn := testAccProvider.Meta().(*FastlyClient).conn
+		honeycombList, err := conn.ListHoneycombs(&gofastly.ListHoneycombsInput{
+			Service: service.ID,
+			Version: service.ActiveVersion.Number,
+		})
+
+		if err != nil {
+			return fmt.Errorf("[ERR] Error looking up Honeycomb Logging for (%s), version (%d): %s", service.Name, service.ActiveVersion.Number, err)
+		}
+
+		if len(honeycombList) != len(honeycomb) {
+			return fmt.Errorf("Honeycomb List count mismatch, expected (%d), got (%d)", len(honeycomb), len(honeycombList))
+		}
+
+		log.Printf("[DEBUG] honeycombList = %#v\n", honeycombList)
```

Comment:
Is this still needed?

Reply:
It could be helpful for inspecting global state in debug logs.